File: haskell/2022/day1.hs
import Data.List (sortBy)
split :: Eq a => (a -> Bool) -> [a] -> [[a]]
split _ [] = []
split pred str = 
    let (start, rest) = break pred str
        (_, remain) = span pred rest
     in start : split pred remain

main :: IO() = do
    content <- readFile "input.txt"
    let res = sortBy (flip compare) . map(sum . map read) . split (=="") . lines $ content :: [Int]
    print $ head res
    print $ sum . take 3 $ res

File: cpp/2018/day1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ifstream f("input.txt");
    int a, res = 0;
    set<int> s;
    vector<int> v;
    while (f >> a) v.push_back(a);
    int sum = 0;
    for (int a : v) sum += a;
    cout << sum << endl;
    int i = 0;
    while (1) {
        res += v[i];
        if (!s.insert(res).second) {
            cout << res;
            return 0;
        }
        (++i) %= v.size();
    }
}